### DISCUSSION TITLE
How to use CustomFilter.sql to filter android special mime_type.

### DISCUSSION THREAD

**Discussion by YanhqZ** [fluttercandies/flutter_photo_manager#927] [Q&A] [answered]
### Version

2.6.0

### Platforms

Android

### Device Model

Galaxy S22Ultra(Android 13)

### flutter info

```sh
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.10, on macOS 12.6.1 21G217 darwin-arm, locale
    zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] IntelliJ IDEA Community Edition (version 2022.3.3)
[✓] Connected device (3 available)
[✓] HTTP Host Availability
```

### How to reproduce?

I use CustomFilter.sql to filter android special mime_type data, but face error. Is there have any solution？

### Logs

```sh
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.10, on macOS 12.6.1 21G217 darwin-arm, locale
    zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] IntelliJ IDEA Community Edition (version 2022.3.3)
[✓] Connected device (3 available)
[✓] HTTP Host Availability
```

### Example code (optional)

```Dart
final List<AssetPathEntity> paths = await PhotoManager.getAssetPathList(
      onlyAll: true,
      type: RequestType.image,
      filterOption: createFilterOption(),
    );


  PMFilter? createFilterOption() {
        return CustomFilter.sql(
          where: '${CustomColumns.android.mediaType} = "image/jpeg"',
        );
  }
```

**Comment by YanhqZ** [comment]
logs: 
E/flutter (17131): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: PlatformException(The getAssetPathList method has an error: Invalid token image/jpeg, java.lang.IllegalArgumentException: Invalid token image/jpeg
E/flutter (17131): 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:172)
E/flutter (17131): 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:142)
E/flutter (17131): 	at android.content.ContentProviderProxy.query(ContentProviderNative.java:481)
E/flutter (17131): 	at android.content.ContentResolver.query(ContentResolver.java:1226)
E/flutter (17131): 	at android.content.ContentResolver.query(ContentResolver.java:1158)
E/flutter (17131): 	at android.content.ContentResolver.query(ContentResolver.java:1114)
E/flutter (17131): 	at com.fluttercandies.photo_manager.core.utils.AndroidQDBUtils.getMainAssetPathEntity(AndroidQDBUtils.kt:93)
E/flutter (17131): 	at com.fluttercandies.photo_manager.core.PhotoManager.getAssetPathList(PhotoManager.kt:44)
E/flutter (17131): 	at com.fluttercandies.photo_manager.core.PhotoManagerPlugin.handleMethodResult(PhotoManagerPlugin.kt:273)
E/flutter (17131): 	at com.fluttercandies.photo_manager.core.PhotoManagerPlugin.access$handleMethodResult(PhotoManagerPlugin.kt:27)
E/flutter (17131): 	at com.fluttercandies.photo_manager.core.PhotoManagerPlugin$onHandlePermissionResult$1.invoke(PhotoManagerPlugin.kt:248)
E/flutter (17131): 	at com.fluttercandies.photo_manager.core.PhotoManagerPlugin$onHandlePermissionResult$1.invoke(PhotoManagerPlugin.kt:246)
E/flutter (17131): 	at com.fluttercandies.photo_manager.core.PhotoManagerPlugin$Companion.runOnBackground$lambda$0(PhotoManagerPlugin.kt:44)
E/flutter (17131): 	at com.fluttercandies.photo_manager.core.PhotoManagerPlugin$Companion.$r8$lambda$vWNc2g2BWEDWg6G41MFWJAaVd4g(Unknown Source:0)
E/flutter (17131): 	at com.fluttercandies.photo_manager.core.PhotoManagerPlugin$Companion$$ExternalSyntheticLambda0.run(Unknown Source:2)
E/flutter (17131): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
E/flutter (17131): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
E/flutter (17131): 	at java.lang.Thread.run(Thread.java:1012)
E/flutter (17131): , {hasAll: true, onlyAll: true, type: 1, option: {type: 1, child: {orderBy: [], containsPathModified: false, where: media_type = "image/jpeg"}}}, null)
E/flutter (17131): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:653:7)
E/flutter (17131): #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:296:18)
E/flutter (17131): <asynchronous suspension>
E/flutter (17131): #2      PhotoManagerPlugin.getAssetPathList (package:photo_manager/src/internal/plugin.dart:54:43)
E/flutter (17131): <asynchronous suspension>

**Comment by CaiJingLong** [comment, marked as answer]
Have two questions:

1. This is a like sql, strings need to be wrapped in single quotes instead of double quotes.
2. `mediaType` has document comments, which represent video, image or audio.

So I think you maybe need `mimeType` :

```dart
Future<void> _reproduct() async {
    PMFilter? createFilterOption() {
      return CustomFilter.sql(
        where: '''${CustomColumns.android.mimeType} = 'image/png' ''',
      );
    }

    final assetCount = await PhotoManager.getAssetCount(
      filterOption: createFilterOption(),
      type: RequestType.all,
    );

    print('assets count: $assetCount');
  }
```


<img width="1057" alt="image" src="https://user-images.githubusercontent.com/14145407/232678282-63e8001e-d974-4fd4-8389-c31a50c9372e.png">


By the way, `mimeType` just support android, you need to handle the logic of iOS by yourself.

**Reply by YanhqZ** [reply]
yes，my meaning is mime_type. Thanks for your reply, my questions has solved
